packages/react/src/ReactElement.js

```diff
@@ -406,6 +406,17 @@ export function createElement(type, config, children) {
       if (props[propName] === undefined) {
         props[propName] = defaultProps[propName];
       }
+      if (__DEV__) {
+        if (propName === 'ref' || propName === 'key') {
+          console.error(
+            '%s: %s is not a valid default prop to set. ' +
```

Comment:
"is not a default prop to set" sounds confusing. Can you come up with a better wording?